cmd/util/cmd/read-protocol-state/cmd/root.go

```diff
@@ -0,0 +1,36 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	flagDatadir string
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "read-protocol-state",
+	Short: "read storage data",
+}
+
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println("error", err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.PersistentFlags().StringVarP(&flagDatadir, "datadir", "d", "/var/flow/data/protocol", "directory to the badger dababase")
+	_ = Cmd.MarkFlagRequired("datadir")
```

Comment:
```suggestion
	_ = rootCmd.MarkPersistentFlagRequired("datadir")
```

I maybe missing something but should this not be attached to the `rootCmd` as all commands that interact with the storage to point to `datadir` and be made `MarkPersistentFlagRequired` as `cobra` will check this first before checking any other required flags.

I guess in this case this doesn't really matter as there's only one command